hindent fails to build with haskell-src-exts >=1.21

## Problem
```
cabal install hindent
```

Log message:
```
Configuring hindent-5.2.7...
Building hindent-5.2.7...
Failed to install hindent-5.2.7
Build log ( /home/exa/work/imcurses/.cabal-sandbox/logs/ghc-8.4.3/hindent-5.2.7-GqIdV7TJaGfJi2BnEdtl5Z.log ):
cabal: Entering directory '/tmp/cabal-tmp-27319/hindent-5.2.7'
Configuring hindent-5.2.7...
Preprocessing library for hindent-5.2.7..
Building library for hindent-5.2.7..
[1 of 4] Compiling HIndent.CabalFile ( src/HIndent/CabalFile.hs, dist/dist-sandbox-ad69a7e0/build/HIndent/CabalFile.o )
[2 of 4] Compiling HIndent.Types    ( src/HIndent/Types.hs, dist/dist-sandbox-ad69a7e0/build/HIndent/Types.o )
[3 of 4] Compiling HIndent.Pretty   ( src/HIndent/Pretty.hs, dist/dist-sandbox-ad69a7e0/build/HIndent/Pretty.o )

src/HIndent/Pretty.hs:453:10: error:
    Duplicate instance declarations:
      instance Pretty Type -- Defined at src/HIndent/Pretty.hs:453:10
      instance Pretty Kind -- Defined at src/HIndent/Pretty.hs:1475:10
    |
453 | instance Pretty Type where
    |          ^^^^^^^^^^^
cabal: Leaving directory '/tmp/cabal-tmp-27319/hindent-5.2.7'
cabal: Error: some packages failed to install:
hindent-5.2.7-GqIdV7TJaGfJi2BnEdtl5Z failed during the building phase. The
exception was:
ExitFailure 1
```

# Cause

This is because haskell-src-exts converted `Kind` from `data Kind l = ...` to `type Kind = Type` in version `1.21`.

# Solution

The obvious immediate workaround is to fix the dependency to explicitly say `<1.21`.

In my very humble opinion, just erasing the `Pretty Kind` instance should work too, but not sure about consequences (the implementations differ, `Kind` has `pretty'`, `Type` has `typ`).

Thanks in advance!

--- a/src/HIndent/Pretty.hs
+++ b/src/HIndent/Pretty.hs
@@ -1143,7 +1143,11 @@ instance Pretty QualConDecl where
                        (pretty d))
 
 instance Pretty GadtDecl where
+#if MIN_VERSION_haskell_src_exts(1,21,0)
+  prettyInternal (GadtDecl _ name _ _ fields t) =
+#else
   prettyInternal (GadtDecl _ name fields t) =
+#endif
     horVar `ifFitsOnOneLineOrElse` verVar
     where
       fields' p =
@@ -1442,8 +1446,10 @@ instance Pretty DataOrNew where
 instance Pretty FunDep where
   prettyInternal = pretty'
 
+#if !MIN_VERSION_haskell_src_exts(1,21,0)
 instance Pretty Kind where
   prettyInternal = pretty'
+#endif
 
 instance Pretty ResultSig where
   prettyInternal (KindSig _ kind) = pretty kind
@@ -1842,6 +1848,9 @@ typ (TyWildCard _ name) =
          pretty n
 typ (TyQuasiQuote _ n s) = quotation n (string s)
 typ (TyUnboxedSum{}) = error "FIXME: No implementation for TyUnboxedSum."
+#if MIN_VERSION_haskell_src_exts(1,21,0)
+typ (TyStar _) = write "*"
+#endif
 
 prettyTopName :: Name NodeInfo -> Printer ()
 prettyTopName x@Ident{} = pretty x

